autopush_rs/src/util/ddb_helpers.rs

```diff
@@ -501,6 +566,241 @@ impl DynamoStorage {
         Box::new(response)
     }
 
+    fn drop_user(
+        ddb: Rc<Box<DynamoDb>>,
+        uaid: &Uuid,
+        router_table_name: &str,
+    ) -> MyFuture<DeleteItemOutput> {
+        let uaid = uaid.simple().to_string();
+        let table_name = router_table_name.to_string();
+        let response = retry_if(
+            move || {
+                ddb.delete_item(&DeleteItemInput {
+                    table_name: table_name.clone(),
+                    key: ddb_item! {
+                        uaid: s => uaid.clone()
+                    },
+                    ..Default::default()
+                })
+            },
+            |err: &DeleteItemError| {
+                matches!(err, &DeleteItemError::ProvisionedThroughputExceeded(_))
```

Comment:
This retry loop I've seen in a number of places, it may be a little more ergonomic to have something like:

```rust
trait Retry {
    fn retry(&self) -> bool;
}

pub fn retrying<F>(f: F) -> impl Future<Item = F::Item, Error = F::Error> 
    where F: Future, F::Error: Retry
{
    retry_if(...)
}

impl Retry for DeleteItemError { ... }
impl Retry forGetItemError { ... }
```